template/src/ontology/Makefile.jinja2

```diff
@@ -552,14 +552,16 @@ PAT=true # Global parameter to bypass pattern generation
 ALL_PATTERN_FILES=$(wildcard $(PATTERNDIR)/dosdp-patterns/*.yaml)
 ALL_PATTERN_NAMES=$(strip $(patsubst %.yaml,%, $(notdir $(wildcard $(PATTERNDIR)/dosdp-patterns/*.yaml))))
 
+PATTERN_CLEAN_FILES=../patterns/all_pattern_terms.txt \
+	$(DOSDP_OWL_FILES_DEFAULT) $(DOSDP_TERM_FILES_DEFAULT) \{% if project.pattern_pipelines_group is defined %}
+	{% for pipeline in project.pattern_pipelines_group.products -%}
+		$(DOSDP_OWL_FILES_{{ pipeline.id.upper() }}) $(DOSDP_TERM_FILES_{{ pipeline.id.upper() }}) {% endfor %}{% endif %}
+
+# Note to future generations: prepending ./ is a safety measure to ensure that 
+# the environment does not malicously set `PATTERN_CLEAN_FILES` to `\`.
 .PHONY: pattern_clean
 pattern_clean:
-	rm -f $(DOSDP_OWL_FILES_DEFAULT)
-	rm -f $(DOSDP_TERM_FILES_DEFAULT){% if project.pattern_pipelines_group is defined %}
-	{%- for pipeline in project.pattern_pipelines_group.products %}
-	rm -f $(DOSDP_OWL_FILES_{{ pipeline.id.upper() }})
-	rm -f $(DOSDP_TERM_FILES_{{ pipeline.id.upper() }})
-	{%- endfor %}{% endif %}
+	rm -f $(patsubst %, ./%, $(PATTERN_CLEAN_FILES))
```

Comment:
cc @gouttegd  I only need you to review these lines (555-564), the rest is fine.